cumulus/zombienet/zombienet-sdk-helpers/src/lib.rs

```diff
@@ -482,19 +537,49 @@ pub async fn assign_cores(
 	Ok(())
 }
 
-pub async fn wait_for_upgrade(
-	client: OnlineClient<PolkadotConfig>,
-	expected_version: u32,
-) -> Result<(), anyhow::Error> {
-	let updater = client.updater();
-	let mut update_stream = updater.runtime_updates().await?;
+/// Creates a runtime upgrade call using `sudo` and `set_code`.
+pub fn create_runtime_upgrade_call(wasm: &[u8]) -> DynamicPayload {
+	zombienet_sdk::subxt::tx::dynamic(
+		"Sudo",
+		"sudo_unchecked_weight",
+		vec![
+			value! {
+				System(set_code { code: Value::from_bytes(wasm) })
+			},
+			value! {
+				{
+					ref_time: 1u64,
+					proof_size: 1u64
+				}
+			},
+		],
+	)
+}
 
-	while let Some(Ok(update)) = update_stream.next().await {
-		let version = update.runtime_version().spec_version;
-		log::info!("Update runtime spec version {version}");
-		if version == expected_version {
-			break;
+/// Wait until a runtime upgrade has happened.
+///
+/// This checks all finalized blocks until it finds a block that sets the
+/// `RuntimeEnvironmentUpdated` digest.
+///
+/// Returns the hash of the block at which the runtime upgrade was applied.
+pub async fn wait_for_runtime_upgrade(
```

Comment:
how about adding `wait_for_runtime_upgrade_with_timeout()` methid that would wrap this one?
It is always wrapped wherever it's called.

Another idea. Maybe we could a combo method `runtime_upgrade()` that would wrap `create_runtime_upgrade_call()` and `wait_for_runtime_upgrade_with[_with_timeout]()`.
Similarly to `assign_cores()`.

Reply:
My point being that CI is flaky. Sometimes an operation takes 3 minutes, sometimes 10 minutes. Depending on the load and whatever. I don't want the CI test to fail because of this. I have seen soo many tests failing because of these random timeouts "yeah something works in X amount of time" and it always broke at some point in CI. 
Also most of the other tests don't use any timeout, why then use a timeout for the runtime upgrade? :D